envoy/router/path_rewrite_policy.h

```diff
@@ -0,0 +1,70 @@
+#pragma once
+
+#include "envoy/config/typed_config.h"
+
+#include "source/common/common/logger.h"
+
+#include "absl/status/statusor.h"
+#include "absl/strings/string_view.h"
+
+namespace Envoy {
+namespace Router {
+
+/**
+ * Used to decide if pattern template rewrite is needed based on the target route.
+ * Subclassing Logger::Loggable so that implementations can log details.
+ */
+class PathRewritePredicate : Logger::Loggable<Logger::Id::router> {
+public:
+  PathRewritePredicate() = default;
+  virtual ~PathRewritePredicate() = default;
+
+  /**
+   * @return the name of the current predicate.
+   */
+  virtual absl::string_view name() const PURE;
+
+  /**
+   * Used to rewrite the current url to the specified output. Can return a failure in case rewrite
+   * is not successful.
+   *
+   * @param current_pattern current url of route
```

Comment:
again if it's the url, maybe call it that?

Reply:
changed